pkg/server/server.go

```diff
@@ -0,0 +1,629 @@
+package server
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/labels"
+
+	k8sClient "k8s.io/kubernetes/pkg/client/unversioned"
+)
+
+//Server struct
+type Server struct {
+	Router *mux.Router
+}
+
+//Global Kubernetes Client
+var client k8sClient.Client
+
+//Init does stuff
+func Init(clientConfig restclient.Config) error {
+	var err error
+	var tempClient *k8sClient.Client
+
+	//In Cluster Config
+	if clientConfig.Host == "" {
+		tempConfig, err := restclient.InClusterConfig()
+		if err != nil {
+			return err
+		}
+		tempClient, err = k8sClient.New(tempConfig)
+
+		client = *tempClient
+
+		//Local Config
+	} else {
+		tempClient, err = k8sClient.New(&clientConfig)
+		if err != nil {
+			return err
+		}
+		client = *tempClient
+	}
+
+	return nil
+}
+
+//NewServer creates a new server
+func NewServer() (server *Server) {
+	router := mux.NewRouter()
+
+	sub := router.PathPrefix("/beeswax/deploy/api/v1").Subrouter()
+
+	sub.Path("/environmentGroups").Methods("GET").HandlerFunc(getEnvironmentGroups)
+
+	sub.Path("/environmentGroups/{environmentGroupID}").Methods("GET").HandlerFunc(getEnvironmentGroup)
```

Comment:
Maybe add 2 routes to get Mux to be more user friendly?  IIRC the only way to get mux to understand

```
GET /environmentGroups/foo
```
and
```
GET /environmentGroups/foo/
``` is to add 2 routes to the same endpoint with the trailing slash.